all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java

```diff
@@ -295,15 +295,29 @@ else if(jsonResponse!=null && jsonResponse.getStatus() == HttpStatus.SC_BAD_REQU
     @Override
     public Response read(Request request) throws BaseException {
         String id=(String)request.getRequest().get(JsonKeys.ID);
-        logger.info("CertServiceImpl:read:idProvided:"+id);
-        Map<String,Object>esCertData=CertificateUtil.getCertificate(id);
-        logger.info("CertServiceImpl:read:esCert data is :"+esCertData);
-        if(MapUtils.isEmpty(esCertData)){
-            throw new BaseException(IResponseMessage.RESOURCE_NOT_FOUND,localizer.getMessage(IResponseMessage.RESOURCE_NOT_FOUND,null), ResponseCode.RESOURCE_NOT_FOUND.getCode());
+        logger.info("CertServiceImpl:read:idProvided: {}",id);
+        Response cassandraResponse = CertificateUtil.getCertRecordByID(id);
+        Response response = new Response();
+        List<Map<String, Object>> resultList = (List<Map<String, Object>>) cassandraResponse.getResult().get(JsonKeys.RESPONSE);
+        if (CollectionUtils.isNotEmpty(resultList) && MapUtils.isNotEmpty(resultList.get(0))) {
+            Map<String, Object> certInfo = resultList.get(0);
+            try {
+                Map<String, Object> data = requestMapper.readValue((String) certInfo.get(JsonKeys.DATA), new TypeReference<Map<String, Object>>() {});
```

Comment:
@aishwa8141 - are we sure that this will maintain the current response json structure?
`"id": "1a7691b1-facc-4047-92de-c5dc73bf00ce",
            "pdfUrl": null,
            "jsonUrl": null,
            "recipient": {
                "name": "3.2user1",
                "email": null,
                "phone": null,
                "id": "247134e0-8661-47a8-9c1e-0a240d5157c1",
                "type": null
            },
            "data": null,
            "related": {
                "type": "sunbird acknowledgement certificate",
                "batchId": "0130929925129666560",
                "courseId": "do_21309298404773068811044"
            },
            "createdBy": null,
            "updatedBy": null,
            "reason": null,
            "accessCode": "J4G9G5",
            "isRevoked": false`

Reply:
yes @indrajra